### ISSUE TITLE
Second thoughts on Platform.select() bindings

### ISSUE THREAD

**Issue by MoOx** [reasonml-community/bs-react-native#316]
[We recently merged a PR that brings `Platform.select` bindings](https://github.com/reasonml-community/bs-react-native/pull/283).
I was going to use it and saw that in practise, it's more limited that Platform.os() function in the ReasonML context (in comparison how it's used in JS, often with the spread operator...)
Let's me illustrate that with a simple example

### Platform.os() usage examples

```re
let styles =
  StyleSheet.create(
    Style.{
      "container":
        switch (Platform.os()) {
        | IOS(_) => style([backgroundColor(String("blue"))])
        | Android => style([backgroundColor(String("red"))])
        | exception (Platform.UnknownPlatform(p)) when p == "web" =>
          style([backgroundColor(String("green"))])
        | exception (Platform.UnknownPlatform(p)) => style([])
        },
      "anotherContainer":
        concat([
          style([width(Pct(80.))]),
          switch (Platform.os()) {
          | IOS(_) => style([backgroundColor(String("blue"))])
          | Android => style([backgroundColor(String("red"))])
          | exception _ => style([])
          },
        ]),
```

Imo that's already the equivalence if Platform.select().
Let's me demonstrate that by converting this code to Platform.select()

### Platform.select() usage examples

```re
let styles =
  StyleSheet.create(
    Style.{
      "container2":
        Platform.select({
          ios: style([backgroundColor(String("blue"))]),
          android: style([backgroundColor(String("red"))]),
          // no way to target web
          // or any other platforms
        }),
      "anotherContainer2":
        concat([
          style([width(Pct(80.))]),
          Platform.select({
            ios: style([backgroundColor(String("blue"))]),
            android: style([backgroundColor(String("red"))]),
            // no way to target other platforms
          }),
        ]),
```

Currently Platform.os() is superior imo 
- because it handles more use cases
- because it fits more the language (by relying on variant)

select() make sense in JS as it's convenient to spread styles (or anything else), but in reasonml, not so sure...

Maybe we could think about another way to bind `select()`, but I am afraid we ends having variants and... it will just look like the `os()` function!

So I was thinking: what about ditching the newly merged `select()` binding + adding more docs about this choice (why we did implement this method) as it's not that useful when taking in consideration the `os()`interface in addition to the language feature.
Example for  `os()`like the one above could be added the docs as an illustration.

Thoughts?

**Comment by SandyChapman** [comment]
You're right that `os()` can do exactly what select does (and more). I prefer the syntax that select provides, however, YMMV.

**Comment by cknitt** [comment]
We shouldn't use `Platform.os()` as implemented in the old bindings because of #239.

For bs-react-native-next, we can revisit providing a `Platform.select` binding or some equivalent of `Platform.os()` that does not suffer of the problem described in #239.

**Comment by MoOx** [comment]
@cknitt I think recent changes fixes #239 right?

**Comment by MoOx** [comment]
Not relevant with current new implementation.
